selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/stack/CreateWorkspaceFromCentosGoStackTest.java

```diff
@@ -0,0 +1,98 @@
+import java.util.List;
+import org.eclipse.che.selenium.core.client.TestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.dashboard.CreateWorkspaceHelper;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.openqa.selenium.By;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Skoryk Serhii */
+public class CreateWorkspaceFromCentosGoStackTest {
+
+  private static final String WORKSPACE_NAME = generate("workspace", 4);
+  private static final String DESKTOP_GO_SIMPLE_PROJECT = "desktop-go-simple";
+  private static final String WEB_GO_SIMPLE_PROJECT = "web-go-simple";
+
+  private List<String> projects =
+      ImmutableList.of(DESKTOP_GO_SIMPLE_PROJECT, WEB_GO_SIMPLE_PROJECT);
+
+  @Inject private Ide ide;
+  @Inject private Consoles consoles;
+  @Inject private Dashboard dashboard;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CreateWorkspaceHelper createWorkspaceHelper;
+  @Inject private TestWorkspaceServiceClient workspaceServiceClient;
+
+  @BeforeClass
+  public void setUp() {
+    dashboard.open();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkWorkspaceCreationFromCentosGoStack() {
+    createWorkspaceHelper.createWorkspaceFromStackWithProjects(CENTOS_GO, WORKSPACE_NAME, projects);
+
+    ide.switchToIdeAndWaitWorkspaceIsReadyToUse();
+
+    projectExplorer.waitProjectInitialization(DESKTOP_GO_SIMPLE_PROJECT);
+    projectExplorer.waitProjectInitialization(WEB_GO_SIMPLE_PROJECT);
+  }
+
+  @Test(priority = 1)
+  public void checkDesktopGoSimpleProjectCommandsStack() {
+    consoles.executeCommandFromProjectExplorer(
+        DESKTOP_GO_SIMPLE_PROJECT,
+        RUN_GOAL,
+        RUN_COMMAND_ITEM.getItem(DESKTOP_GO_SIMPLE_PROJECT),
+        "Hello, world. Sqrt(2) = 1.4142135623730951");
+  }
+
+  @Test(priority = 1)
+  public void checkWebGoSimpleProjectCommandsStack() {
+    consoles.executeCommandFromProjectExplorer(
+        WEB_GO_SIMPLE_PROJECT, RUN_GOAL, RUN_COMMAND, "listening on");
+    consoles.checkWebElementVisibilityAtPreviewPage(
+        By.xpath("//pre[contains(text(),'Hello there')]"));
+    consoles.closeProcessTabWithAskDialog(RUN_COMMAND);
+
+    consoles.executeCommandFromProjectExplorer(
+        WEB_GO_SIMPLE_PROJECT,
+        RUN_GOAL,
+        RUN_COMMAND_ITEM.getItem(WEB_GO_SIMPLE_PROJECT),
+        "listening on");
+    consoles.checkWebElementVisibilityAtPreviewPage(
+        By.xpath("//pre[contains(text(),'Hello there')]"));
```

Comment:
I would suggest to extract string `"//pre[contains(text(),'Hello there')]"` into constant or variable to avoid duplication and to describe proposal of the string.